journal/202308.md

```diff
@@ -0,0 +1,870 @@
+- Added a new integrity check that updates revoked tickets in the database to be either [expired or missed](https://github.com/decred/vspd/pull/415). Initial version of vspd was unable to distinguish between expired and missed tickets and reported them all as "revoked". Knowing how many tickets were missed by the VSP is useful for [evaluating its reliability](https://github.com/decred/vspd/issues/268). This update uses a clever heuristic to guess ticket outcome based on how long it took to revoke it. It is not 100% accurate but is good enough for the purpose of reporting VSP stats.
+
+Developer and internal changes:
+
+- Added test to ensure the hard-coded [error string](https://github.com/decred/vspd/pull/404) returned from [securecookie](https://github.com/gorilla/securecookie) will not change unexpectedly when vspd is upgraded to a newer version of the library.
+- [Replaced](https://github.com/decred/vspd/pull/401) uses of `interface{}` with `any`, similar to [dcrwallet](#dcrwallet).
+- Updated to [Go 1.21](https://github.com/decred/vspd/pull/402) and newer GitHub Actions. Testing with Go 1.19 has been removed.
+- Removed client-side [VSP fee payment logic](https://github.com/decred/vspd/pull/403) from vspd repository. The [idea](https://github.com/decred/vspd/issues/379) to migrate all VSP client code from dcrwallet to vspd repository was found to be problematic, it created a circular dependency between vspd and dcrwallet which is [cumbersome](https://matrix.to/#/!zefvTnlxYHPKvJMThI:decred.org/$_rzNyvnGedWeh_tnXWk5WDQKaj3TODeXbxRa9fg7CtY) and not worth the maintenance burden. Most VSP client code has been removed from vspd and will continue to live in dcrwallet repository. Eventually it should become an exported package which can be reused by software seeking to add VSP staking features.
+  - { _^ is it correct? What client VSP code remains in vspd then? And which module/package consumers like DEX or Cryptopower should use now to not copy-paste dcrwallet's VSP code?_ }
+- Simplified [shutdown code](https://github.com/decred/vspd/pull/410). It was inherited from dcrwallet and was a bit more flexible and complex than required for vspd.
+- Various [refactoring](https://github.com/decred/vspd/pull/408) to allow more code reuse and improve integrity checks.
```

Comment:
408 409 and 412 were just intermediate steps towards getting the GCS work in. I don't think they're very noteworthy, but keep them if you like.